scripts/bootloader/zb_ota/package.py

```diff
@@ -0,0 +1,446 @@
+        manifest = self.create_manifest()
+
+        with open(os.path.join(self.work_dir, Package.MANIFEST_FILENAME), "w") as manifest_file:
+            manifest_file.write(manifest)
+
+        # Package the work_dir to a zip file
+        Package.create_zip_package(self.work_dir, filename)
+
+        # Delete the temporary directory
+        self.rm_work_dir(preserve_work_dir)
+
+    @staticmethod
+    def __create_temp_workspace():
+        return tempfile.mkdtemp(prefix="nrf_dfu_pkg_")
+
+    @staticmethod
+    def create_zip_package(work_dir, filename):
+        files = os.listdir(work_dir)
+
+        with ZipFile(filename, 'w') as package:
+            for _file in files:
+                file_path = os.path.join(work_dir, _file)
+                package.write(file_path, _file)
+
+    @staticmethod
+    def calculate_file_size(firmware_filename):
+        b = os.path.getsize(firmware_filename)
+        return b
+
+    @staticmethod
+    def calculate_sha256_hash(firmware_filename):
+        read_buffer = 4096
+
+        digest = hashlib.sha256()
+
+        with open(firmware_filename, 'rb') as firmware_file:
+            while True:
+                data = firmware_file.read(read_buffer)
+
+                if data:
+                    digest.update(data)
+                else:
+                    break
+
+        # return hash in little endian
+        sha256 = digest.digest()
+        return sha256[31::-1]
+
+    @staticmethod
+    def calculate_crc(crc, firmware_filename):
+        """
+        Calculates CRC16 has on provided firmware filename
+
+        :type str firmware_filename:
+        """
+        data_buffer = b''
+        read_size = 4096
+
+        with open(firmware_filename, 'rb') as firmware_file:
+            while True:
+                data = firmware_file.read(read_size)
+
+                if data:
+                    data_buffer += data
+                else:
+                    break
+        if crc == 16:
+            return calc_crc16(data_buffer, 0xffff)
+        elif crc == 32:
+            return binascii.crc32(data_buffer)
+        else:
+            raise ValueError("Invalid CRC type")
+
+
+    def create_manifest(self):
+        manifest = ManifestGenerator(self.firmwares_data)
+        return manifest.generate_manifest()
+
+    def __add_firmware_info(self, firmware_type, firmware_version, filename, init_packet_data, sd_size=None, bl_size=None):
+        self.firmwares_data[firmware_type] = {
+            FirmwareKeys.FIRMWARE_FILENAME: filename,
+            FirmwareKeys.INIT_PACKET_DATA: init_packet_data.copy()
+            }
+
+        if firmware_version is not None:
+            self.firmwares_data[firmware_type][FirmwareKeys.INIT_PACKET_DATA][PacketField.FW_VERSION] = firmware_version
+
+    @staticmethod
+    def normalize_firmware_to_bin(work_dir, firmware_path):
+        firmware_filename = os.path.basename(firmware_path)
+        new_filename = firmware_filename.replace(".hex", ".bin")
```

Comment:
In theory there can be `.hex` strings inside filename (not in the end of a file) and these occurrences will be also replaced

Reply:
fixed